### PR TITLE
sql: add eventlog for UDF relevant statements

### PR DESCRIPTION
**Pull request by chengxiong-ruan**
Backport addresses #86058.

Release note: None
Release justification: low risk GA blocker

### CODE REVIEW

File: pkg/sql/alter_function.go [hunk 1]
```diff
@@ -86,7 +87,26 @@ func (n *alterFunctionOptionsNode) startExec(params runParams) error {
...
+		}
+		fctx.FormatNode(n.n.Options[i])
+	}
+	fnName, err := params.p.getQualifiedFunctionName(params.ctx, fnDesc)
+	if err != nil {
+		return err
+	}
+	event := eventpb.AlterFunctionOptions{
+		FunctionName: fnName.FQString(),
+		Options:      fctx.CloseAndGetString(),
```
Comment (chengxiong-ruan) [inline pkg/sql/alter_function.go:107]: adding `alter function` details here is probably a overkill....

**Comment by postamar** [comment]
Why are there no tests?

**Comment by chengxiong-ruan** [comment]
> Why are there no tests?

oops, my bad, adding it now

File: pkg/sql/alter_function.go (hunk 1 above)
Comment (chengxiong-ruan) [reply pkg/sql/alter_function.go]: the event log json records the whole statement, so no need to book keeping redundant option details

**Comment by chengxiong-ruan** [comment]
@postamar logic tests added

**Review by postamar — APPROVED** [review]
This LGTM but I don't have strong opinions about ddl event logging so you might want to solicit another review from someone who does.

**Comment by chengxiong-ruan** [comment]
shipping it anyway. don't think it'd be a big deal as long as the event type are clear, desc id and ddl statement are well recorded.
TFTR!
bors r+

**Pull request merged**